### ISSUE TITLE
`request.node.get_marker('name')` returns `None` for markers applied in classes

### ISSUE THREAD

**Issue by norbertpy** [pytest-dev/pytest#902]
I don't know why, but I can't get the marker from my fixture. The `request.node.get_marker()` seems to return `None` regardless of the marker applied to class:

``` python
    @pytest.fixture(scope='class')
    def my_fixture(request):
        marker = request.node.get_marker('joe') # Is always None


    @pytest.mark.joe
    class TestMarker(object):

        def test_marker(self, my_fixture):
            assert True
```

I even registered the marker in my `pytest.ini` file:

``` ini
    # content of pytest.ini
    [pytest]
    markers =
        joe: mark a test as joejoejoe
```

Is this a bug or I'm missing something? Not to mention, I'm using `2.7.2` version.
## Update 1:

If I get rid of the class and apply the marker to a function, it'll work:

``` python
@pytest.mark.joe
def test_marker(my_fixture):
    assert True
```
## Update 2:

This actually works:

``` python
@pytest.fixture(scope='class')
def my_fixture(request):
    marker = next((m for m in request.cls.pytestmark if m.name == 'joe'), None)
```

**Comment by nicoddemus** [comment]
I could reproduce the problem in `master`, thanks for reporting it.

It seems related to markers applied to classes (I updated the title to match this).

**Comment by norbertpy** [comment]
@nicoddemus , thank you. Do I get notified if it's fixed?

**Comment by nicoddemus** [comment]
Yes, you're watching this thread because you submitted this issue. :smile:

Thanks again.

**Comment by RonnyPfannschmidt** [comment]
this issue is related to marker transfers

i believe the most correct fix would be implementing #891 however it comes with compatibility issues

**Comment by The-Compiler** [comment]
Removing the milestone here, as this shouldn't block 3.0 and we have #1670 (where I set the milestone to 4.0 for now) to track all marker staining related issues.

**Comment by RonnyPfannschmidt** [comment]
this one can be solved after #2522

**Comment by RonnyPfannschmidt** [comment]
fixed in #3317
